moped-editor/src/layouts/DashboardLayout/NavBar/NavigationSearchInput.js

```diff
@@ -0,0 +1,246 @@
+}));
+
+/**
+ * @return {JSX.Element}
+ * @constructor
+ */
+const NavigationSearchInput = () => {
+  const classes = useStyles();
+  const divRef = React.useRef();
+  let projectSearchQuery = new GQLAbstract(ProjectsListViewQueryConf);
+
+  // Toggle Text Input or magnifying glass
+  const [searchInput, showSearchInput] = useState(false);
+  // anchor element for results popper to "attach" to
+  const [searchResultsAnchor, setSearchResultsAnchor] = useState(null);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const [loadSearchResults, { called, loading, data }] = useLazyQuery(
+    projectSearchQuery.gql,
+    projectSearchQuery.config.options.useQuery
+  );
+
+  const handleMagClick = () => {
+    showSearchInput(true);
+  };
+
+  const handleDropdownClose = () => {
+    setSearchResultsAnchor(null);
+    showSearchInput(false);
+    setSearchTerm("");
+  };
+
+  const handleSearchFocus = () => {
+    setSearchResultsAnchor(divRef.current);
+  };
+
+  /**
+   * Handles special keys typed in the search bar
+   * @param {string} key - The key name being typed
+   */
+  const handleKeyDown = key => {
+    switch (key) {
+      // On Escape key, clear the search
+      case "Escape":
+        setSearchTerm("");
+        setSearchResultsAnchor(null);
+        break;
+      // On Enter key, initialize the search
+      case "Enter":
+        handleSearchSubmission(null);
+        break;
+
+      default:
+        return;
+    }
+  };
+
+  /**
+   * Formats search term and triggers lazy query to load search results
+   * @param {event} - Event that triggered submission
+   */
+  const handleSearchSubmission = event => {
+    // Stop if we don't have any value entered in the search field
+    if (searchTerm.length === 0) {
+      console.log("no search term");
+    }
+
+    // Prevent default behavior on any event
+    if (event) event.preventDefault();
+
+    // Formats search query based on project search columns and config
+    Object.keys(projectSearchQuery.config.columns)
+      .filter(column => projectSearchQuery.config.columns[column]?.searchable)
+      .forEach(column => {
+        const {
+          operator,
+          quoted,
+          envelope,
+        } = projectSearchQuery.config.columns[column].search;
+        const searchValue = getSearchValue(
+          projectSearchQuery,
+          column,
+          searchTerm
+        );
+        const graphqlSearchValue = quoted
+          ? `"${envelope.replace("{VALUE}", searchValue)}"`
+          : searchValue;
+
+        projectSearchQuery.setOr(column, `${operator}: ${graphqlSearchValue}`);
+      });
+
+    // Initiate Lazy Query to get search results
+    loadSearchResults();
+  };
+
+  // Search as user updates searchTerm
+  useEffect(() => {
+    const delayDebounceFn = setTimeout(() => {
+      handleSearchSubmission(null);
+    }, 500);
+
+    return () => clearTimeout(delayDebounceFn);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchTerm]);
```

Comment:
🙏 Thank you!

Reply:
i'm a little worried that i'm not supposed to use the eslint disable next line thing here, but it seems like this is its intended use

Reply:
@chiaberry As long as you tried adding all the dependencies and for some reason it still didn't work, it's ok. Personally, I am a little too liberal of its use myself, usually end up using it whenever I am trying to avoid an infinite loop situation.